client/src/components/Memories.js

```diff
@@ -0,0 +1,186 @@
+    const requestBody = {
+      comment: comment
+    }
+    try {
+      const response = await handlePost(endpoint, requestBody)
+      if(response.status === 200) {
+        setOpenSnackbar(true);
+        setSnackbarMessage("Comment added successfully!")
+        setTimeout(() => {
+          setOpenSnackbar(false);
+          setSnackbarMessage("")
+          //clear comment form on comment submit success
+          setNewComment({
+            memory_uuid: "",
+            user_uuid: "",
+            comment_text: "",
+          });
+          getMemories()
+        }, 1500)
+                
+      } else {
+        alert("Comment could not be saved, try again.")
+      }
+    } catch {
+      alert("comment could not be added.")
+    }
+  } 
+  if(memories.length === 0) {
+    return (
+      <div className="Memories">
+        <Snackbar open={openSnackbar} autoHideDuration={1500} message={snackbarMessage} anchorOrigin={{horizontal: "center", vertical:"top"}}/>
+        <section className='memories-grid'>
+          <h4>No memories yet.</h4>
+        </section>
+      </div>
+    );
+  } else {
+    console.log(memories)
+    return (
+      <div className="Memories">
+        <Snackbar open={openSnackbar} autoHideDuration={1500} message={snackbarMessage} anchorOrigin={{horizontal: "center", vertical:"top"}}/>
+        <section className='memories-grid'>
+          {memories.map(m => {
+            return <MemoryTile m={m} likePost={likePost} submitComment={submitComment}/>
+          })}
+        </section>
+      </div>
+    );
+  }
+ 
+}
+
+
+const MemoryTile = (props) => {
+  const m = props.m;
+  const likePost = props.likePost;
+  const [likeDisabled, setLikeDisabled] = useState(false)
+  const [showComments, setShowComments] = useState(false); //show and hide comment view
+  const submitComment = props.submitComment;
+  const handleClick = () => {
+    likePost(m);
+    setLikeDisabled(!likeDisabled)
+  }
+  if(m.num_likes === 0) {
+    return (
+      <section className='memory'>
+        <h4>Name: {m.name}</h4>
+        <h4>Special Occasion: {m.occasion}</h4>
+        <h4>Experience: {m.experience}</h4>
+        <h4>Likes: {m.num_likes}</h4>
+        <button onClick = {handleClick}>+</button>
+      </section>
+    )
+  } else {
+    return (
+      <section className='memory'>
+        <h4>Name: {m.name}</h4>
+        <h4>Special Occasion: {m.occasion}</h4>
+        <h4>Experience: {m.experience}</h4>
+        <h4>Likes: {m.num_likes}</h4>
+        <button onClick = {handleClick} disabled={likeDisabled}>+</button>
+        <button onClick ={() => setShowComments(!showComments)}>{showComments ? `Hide ${String.fromCharCode(8593)}` : `View Comments (${m.comments.length}) ${String.fromCharCode(8595)}`}</button>
+        {showComments ? 
+        <ViewComments
+          m = {m}
+          submitComment={submitComment} 
+        /> 
+        : null}
+      </section>
+    )
+  }
+}
+
+
+
+const ViewComments = (props) => {
```

Comment:
same here. Should be a standalone file

Reply:
Should child components get their own file?

Reply:
Done!